### PR TITLE
[WIP] Add button to hide resource previews

### LINKED ISSUE #6542: A way to turn off the material preview in the Inspector?
**Operating system or device - Godot version:**
Godot 2.1, any OS.

**Issue description** (what happened, and what was expected):
_This is a feature request._

I don't really like the new shader material preview in the inspector, I am already previewing my materials in the viewport and it's kind of a waste of space in the inspector, IMO.

Illustration

Since I couldn't find an option for it:
Would it be possible to get an option to turn it off in the Engine Settings > Inspector Tab.

:sparkles: Pretty please? :sparkles:

### PR DESCRIPTION
**Pull request by GalanCM**
fixes #6542 (but not the lag problem mentioned in the discussion).

Since Juan is currently reworking the resources, it seems most don't currently have previews, so it's currently only implemented for textures.

There's also some room for improvements, but since this is my first serious pull request, and I'm not exactly comfortable working in C++, I thought I'd make sure I'm on the right track before tackling any more.

### CODE REVIEW

File: tools/editor/resource_preview_dock.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/*************************************************************************/
+/*  texture_editor_plugin.cpp                                            */
```
Comment (bojidar-bg) [inline tools/editor/resource_preview_dock.cpp]: Slight issue, you forgot to change the filename :smiley:

Comment (GalanCM) [inline tools/editor/resource_preview_dock.cpp]: Thanks for the catch. I'll make sure to fix that before my next push.

**Comment by bojidar-bg** [comment]
Bump! This PR is getting stale.

**Comment by GalanCM** [comment]
Sorry, I was waiting for some of the missing resource previews (materials and audio) to be restored. I'm in the last two weeks before releasing my first commercial game right now, but once that's done, I can try to polish this up to something a bit more merge-worthy.

**Comment by bojidar-bg** [comment]
@GalanCM Ok, no pressure! Good luck with releasing your game, and may the Godot be with you :smile:

**Comment by akien-mga** [comment]
I guess the master branch should be ready for this to be rebased now.

**Comment by akien-mga** [comment]
> I guess the master branch should be ready for this to be rebased now.

Ping.

**Comment by GalanCM** [comment]
I've been looking at updating this. I feel like the code base (not to mention UI) has diverged so much that it would probably be easier to just redo this on a fresh branch. I'm going to try to start working on it in the next few days.

**Comment by akien-mga** [comment]
Closing it for now to decrease the backlog of stale PRs, but don't hesitate to open a new PR once ready for merge.

**Pull request closed without merging**